tests/test_exacttarget_bookmarks.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+from dateutil.parser import parse
+from base import ExactTargetBase
+import tap_tester.connections as connections
+import tap_tester.menagerie as menagerie
+import tap_tester.runner as runner
+import datetime
+import dateutil.parser
+import pytz
+
+class ExactTargetBookmarks(ExactTargetBase):
+    def name(self):
+        return "tap_tester_exacttarget_bookmarks"
+
+    def convert_state_to_utc(self, date_str):
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return datetime.datetime.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
+    def test_run(self):
+        self.run_test(self.streams_to_select() - {'data_extension.test 1'}, '2019-01-01T00:00:00Z')
+        self.run_test({'data_extension.test 1'}, '2021-08-01T00:00:00Z')
+
+    def run_test(self, streams, start_date):
+        self.START_DATE = start_date
+
+        conn_id = connections.ensure_connection(self, original_properties=False)
+        runner.run_check_mode(self, conn_id)
+
+        expected_streams = streams
+
+        found_catalogs_1 = menagerie.get_catalogs(conn_id)
+        self.select_found_catalogs(conn_id, found_catalogs_1, only_streams=expected_streams)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        ### Update State
+        ##########################################################################
+
+        new_state = {'bookmarks': dict()}
+        replication_keys = self.expected_replication_keys()
+        for stream in expected_streams:
+            if self.is_incremental(stream):
+                new_state['bookmarks'][stream] = dict()
+                new_state['bookmarks'][stream]['field'] = next(iter(replication_keys[stream]))
+                new_state['bookmarks'][stream]['last_record'] = '2021-08-23T00:00:00Z' if stream == 'data_extension.test 1' else '2019-01-14T00:00:00Z'
+
+        # Set state for next sync
+        menagerie.set_state(conn_id, new_state)
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # Run a sync job using orchestrator
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        for stream in expected_streams:
+            # skip "subscriber" stream as replication key is not retrievable
+            if stream == "subscriber":
+                continue
+
```

Comment:
Can you elaborate on why this stream cannot be tested. If we cannot get a replication key value from the api, we should consider making this stream Full Table.

Reply:
I am still not clear on why this stream cannot be tested here.

Reply:
`Subscriber` (child stream) will not have any bookmark as it only syncs the provided list of subscribers keys and the subscribers key will be created from the `list_subscribers` (parent stream) based on the records returned from the API.

As discussed [here](https://github.com/singer-io/tap-exacttarget/pull/68#discussion_r703402475)